recipes/itk/all/conanfile.py

```diff
@@ -0,0 +1,229 @@
+        cmake.definitions["Module_ITKNeuralNetworks"] = "ON"
+        cmake.definitions["Module_ITKOptimizers"] = "ON"
+        cmake.definitions["Module_ITKOptimizersv4"] = "ON"
+        cmake.definitions["Module_ITKPDEDeformableRegistration"] = "ON"
+        cmake.definitions["Module_ITKPath"] = "ON"
+        cmake.definitions["Module_ITKPolynomials"] = "ON"
+        cmake.definitions["Module_ITKQuadEdgeMeshFiltering"] = "ON"
+        cmake.definitions["Module_ITKRegionGrowing"] = "ON"
+        cmake.definitions["Module_ITKRegistrationCommon"] = "ON"
+        cmake.definitions["Module_ITKRegistrationMethodsv4"] = "ON"
+        cmake.definitions["Module_ITKReview"] = "ON"
+        cmake.definitions["Module_ITKSignedDistanceFunction"] = "ON"
+        cmake.definitions["Module_ITKSmoothing"] = "ON"
+        cmake.definitions["Module_ITKSpatialFunction"] = "ON"
+        cmake.definitions["Module_ITKThresholding"] = "ON"
+        cmake.definitions["Module_ITKVideoCore"] = "ON"
+        cmake.definitions["Module_ITKVideoFiltering"] = "ON"
+        cmake.definitions["Module_ITKVideoIO"] = "OFF"
+        cmake.definitions["Module_ITKVoronoi"] = "ON"
+        cmake.definitions["Module_ITKWatersheds"] = "ON"
+        cmake.definitions["Module_ITKDICOMParser"] = "ON"
+
+        cmake.definitions["Module_ITKVTK"] = "OFF"
+        cmake.definitions["Module_ITKVtkGlue"] = "OFF"
+
+        cmake.definitions["USE_FFTWF"] = "OFF"
+        cmake.definitions["USE_FFTWD"] = "OFF"
+
+        # Disabled on Linux (link errors)
+        cmake.definitions["Module_ITKLevelSetsv4Visualization"] = "OFF"
+
+        # Disabled because Vxl vidl is not build anymore
+        cmake.definitions["Module_ITKVideoBridgeVXL"] = "OFF"
+
+        cmake.configure(build_folder=self._build_subfolder)
+        return cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.configure(build_folder=self._build_subfolder)
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "gdcmopenjpeg-2.1"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "etc"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "cmake"))
+
+    def package_info(self):
+
+        # in linux we need to link also with these libs
+        if self.settings.os == "Linux":
+            self.cpp_info.libs.extend(["pthread", "dl", "rt"])
+
+        self.cpp_info.builddirs = ['lib/cmake/ITK-{0}/'.format(
+            self._upstream_version)]
+        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.includedirs = ['include']
```

Comment:
Isn't that already the default?

Reply:
maybe. Anyway its not the priority on this, which does not compile on all targets